doc/howtos/using_ubuntu_as_sos.rst

```diff
@@ -0,0 +1,184 @@
+        i915.tsd_init=7 i915.tsd_delay=2000 i915.avail_planes_per_pipe=0x00000F i915.domain_plane_owners=0x011111110000 
+        i915.enable_guc_loading=0 i915.enable_guc_submission=0 i915.enable_preemption=1 i915.context_priority_mode=2 
+        i915.enable_gvt=1 hvlog=2M@0x1FE00000
+}
+Note: here you have to use PARTUUID, UUID doesn’t work.  
+
+Step #12,  Update grub
+# sudo update-grub
+At this moment, you need to modify /boot/grub/grub.cfg file manually to enable the timeout so that the system has opportunity to show 
+you grub menu to select. Or you need to set the “ACRN ubuntu SOS” as the default boot option. Without enabling  timeout, the system 
+will not show grub menu for people to select and boot into incorrect kernel.
+
+Enable timeout:
+ if [ x$feature_timeout_style = xy ] ; then
+ -  set timeout_style=hidden
+ + #set timeout_style=hidden
+ -  set timeout=0
+ + set timeout = 10 
+ # Fallback hidden-timeout code in case the timeout_style feature is
+  # unavailable.
+
+Step #13, Check the boot process 
+Reboot system to see if the acrn.efi is loaded successfully, and if the Menu of Grub being shown to select “ACRN ubuntu SOS”.  
+after boot successfully, the system will start to Ubuntu Desktop successfully.  to check if the system is running on virtualization 
+evn and the ACNR hypervisor is running. 
+For example:
+   
+NUC6CAYH:~$ dmesg | grep ACRN
+[    0.000000] Hypervisor detected: ACRN
+[    0.862942] ACRN HVLog: acrn_hvlog_init
+
+This output means the boot flow is correct and the SOS system with ACRN is running well.
+ 
+
+Step #14, Deploy Clear Linux as the UOS 
+Download clear linux UOS from download.clearlinux.org
+#cd /root
+# wget https://download.clearlinux.org/releases/22780/clear/clear-22780-kvm.img.xz
+# unxz clear-22780-kvm.img.xz
+Copy  launch_uos.sh from source code
+# cp ~/Download/acrn-hypervisor/devicemodel/samples/nuc/launch_uos.sh /root
+Download standard PK kernel with below command:
+# wget https://download.clearlinux.org/releases/22780/clear/x86_64/os/Packages/linux-pk414-standard-4.14.47-44.x86_64.rpm
+# rpm2cpio linux-pk414-standard-4.14.47-44.x86_64.rpm | cpio -idmv
+
+Use the following steps to update clearlinux kernel:
+# sudo losetup -f -P --show /root/clear-22789-kvm.img  
+# sudo mount /dev/loop0p3 /mnt
+# sudo cp -r /root/usr/lib/modules/4.14.47-44.pk414-standard /mnt/lib/modules/
+# sudo cp -r /root/usr/lib/kernel /lib/modules/
+# sudo umount /mnt
+# sync
+
+You may Need to a permission issue
+# sudo chmod 777 /dev/acrn_vhm
+You also need to install a package.
+# sudo apt-get instal iasl
+# sudo cp /usr/bin/iasl /usr/sbin/iasl
+
+Step 15, Deploy device module - acrn-dm
+# cp ~/download/acrn-hypervisor//build/devicemodel/acrn-dm /usr/bin
+
+Step 16, modify launch_uos.sh 
+-s 3,virtio-blk,/root/clear-22780-kvm.img
+-k /lib/modules/kernel/org.clearlinux.pk414-standard.4.14.47-44
```

Comment:
I think this is the reason why you copied stuff from the linux PK standard package on the Service OS rootfs. I would actually prefer if we kept the rootfs 'clean' and make the `launch_uos.sh` script pick the kernel from `/root/...` directly (so we can also save the copy operation above)

Reply:
yes, you got it,
it's hard to implement what said now because clear team provides kvm,img for user OS, if there is a UOS.img provided from Clear team, that would be easy for us to save the copy operation

Reply:
I don't understand what is hard about this. You could just adjust the `-k` parameter in the launch script to point at the kernel in `/root/usr/lib/kernel`? Any reason why this wouldn't work?